### PR TITLE
feat: extend Kubernetes AWS auth to allow specifying `clusterName` in addition to `name/displayName`

### PR DESCRIPTION
**Pull request by cyrus-mc**
## Hey, I just made a Pull Request!

Pull request to address https://github.com/backstage/backstage/issues/15668. When configuring EKS clusters it helps to have the display name differ from the cluster name used to generate bearer token.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by cyrus-mc** [comment]
Fairly new to the Backstage ecosystem. Not exactly sure how I can test this locally. The change is pretty trivial but as of yet I haven't validated that it works as expected.

File: plugins/kubernetes-backend/src/kubernetes-auth-translator/AwsIamKubernetesAuthTranslator.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export class AwsIamKubernetesAuthTranslator
     );
 
     clusterDetailsWithAuthToken.serviceAccountToken = await this.getBearerToken(
-      clusterDetails.name,
```
Comment (cyrus-mc) [inline plugins/kubernetes-backend/src/kubernetes-auth-translator/AwsIamKubernetesAuthTranslator.ts:135]: Maintains backwards compatibility. If `clusterName` isn't supplied default to the original `name` setting.

**Comment by sydorovdmytro** [comment]
I'm a bit curious what is the difference between this PR and 2 rejected ones? 
https://github.com/backstage/backstage/pull/14681
https://github.com/backstage/backstage/pull/14712

**Comment by mclarke47** [comment]
Thanks for the PR! 

As @sydorovdmytro pointed out this seems similar to some previous PRs that were closed because we think we need an RFC. Perhaps you could get involved with these other contributors and see how we could fix this?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Just want to echo @mclarke47. I think we need look for a less patchy solution to this and have something that just works out of the box.

**Pull request closed without merging**